test/tests.py

```diff
@@ -544,6 +543,19 @@ def verify_dmlc_messages(self, stderr, expected_msgs):
             self.expect_messages(
                 kind, actual_msgs.get(kind, []), expected_msgs.get(kind, []))
 
+def _unittest_parse_messages():
+    stderr = '''
+/tmp/f.dml:3:20: In template abc
+/tmp/g.dml:4:9: error EDPARAM: blah blah
+/tmp/f.dml:8:12: conflicting definition
+'''.splitlines()
+    assert (list(DMLFileTestCase.parse_messages(stderr))
+            == [('error', 'EDPARAM',
+                 [('g.dml', 4, "/tmp/g.dml:4:9: error EDPARAM: blah blah"),
+                  ('f.dml', 3, "/tmp/f.dml:3:20: In template abc"),
+                  ('f.dml', 8, "/tmp/f.dml:8:12: conflicting definition")])])
+_unittest_parse_messages()
```

Comment:
Why not a proper test case?

Reply:
Test cases are for testing DMLC. This one tests the test.

Reply:
Ah.